JSIL/ILBlockTranslator.cs

```diff
@@ -1749,6 +1749,23 @@ bool forDynamic
                     }
                 }
             }
+            else if (
```

Comment:
This is a little awkward but I'm willing to merge it.

An explanatory comment would be great, and maybe reduce the redundancy of the if conditions. I think most of the stuff in the 'else if' condition here is redundant with the conditions inside of that else if block.

Both of those improvements can come later though.